### PR TITLE
Implement filter "All upcoming"

### LINKED ISSUE #5921: Search: Fix and Enhance filter options
The filters are not consistent, e.g. it is currently not possible to choose all location types at the same time even though this is the default search behavior. Therefore please make the filter display and search functionality consistent and allow searching of past events as well. 

Please change the filters as follows and make the below options the standard search setting:
- Rename "Event Location Type" to "Setting"
  - Tick all boxes Online, Location, Mixed - by default
- If user only ticks online show all online events already even if no event name has been specified
- Date 
   - Change "All dates" filter to actually display "All dates" including past dates and move it to the end of the search filter area
   - Implement filter "Past" and "All Upcoming"
   - By default tick "All Upcoming" and when ticking "All Upcoming" tick "Today" and all other future filters (except custom dates)
   - Show "Past" at the end of the widget before "All dates"
- Categories - leave it as is to "All Categories"
- Event Types - leave it as is to "All Types"
- Ticket Type - tick Free and Paid by default
- Event Call For Speakers rename it to "Call for Speakers"
- By default tick "Open" only
- Finally change "Clear Filters" to "Default Filters" and reset filters to the standard search filter config when the user clicks it

Furthermoe
* Ensure that the search as above works in the same way from the start page as well the explore page
- The search from the start page should always use the standard search parameters for now

Screenshot from 2020-12-05 13-30-59

### PR DESCRIPTION
**Pull request by Soumi150**
Fixes #5921

#### Short description of what this resolves:
Added  filter 'All Upcoming' to explore page

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/routes/explore.js [hunk 1]
```diff
@@ -252,6 +252,21 @@ export default class ExploreRoute extends Route {
           }
         ]
       });
+    } else if (params.is_upcoming) {
+      filterOptions.push({
+        and: [
+          {
+            name : 'starts-at',
+            op   : 'lt',
```
Comment (maze-runnar) [inline app/routes/explore.js:260]: it should be `gt` here

File: app/routes/explore.js [hunk 2]
```diff
@@ -252,6 +252,21 @@ export default class ExploreRoute extends Route {
...
+            val  : moment().toISOString()
+          },
+          {
+            name : 'ends-at',
+            op   : 'lt',
+            val  : moment().toISOString()
+          }
```
Comment (maze-runnar) [inline app/routes/explore.js:267]: what is need of `ends-at` in upcoming

File: app/routes/explore.js (hunk 1 above)
Comment (Soumi150) [reply app/routes/explore.js]: Yes ,I made it wrong while fixing indentation . Thanks

File: app/routes/explore.js (hunk 2 above)
Comment (Soumi150) [reply app/routes/explore.js]: Removed it.

File: app/routes/explore.js [hunk 3]
```diff
@@ -252,6 +252,16 @@ export default class ExploreRoute extends Route {
           }
         ]
       });
+    } else if (params.is_upcoming) {
+      filterOptions.push({
+        and: [
```
Comment (MeghalBisht) [inline app/routes/explore.js:257]: why and?

File: app/templates/components/explore/side-bar.hbs [hunk 4]
```diff
@@ -96,12 +96,19 @@
...
+                @checked={{if (eq this.is_upcoming "true") true}}
+                @onChange={{action "enableUpcomingEvents" "true"}} />
+            </div>
             <UiCheckbox
               @label={{t "Past"}}
               @class="ui checkbox"
               @checked={{if (eq this.is_past "true") true}}
               @onChange={{action "enablePastEvents" "true"}} />
-          </div>
+             </div>
```
Comment (MeghalBisht) [inline app/templates/components/explore/side-bar.hbs:111]: Check the indentation here

**Review by sachinchauhan2889 — CHANGES_REQUESTED** [review]
Please understand desired result carefully. This is not the desired result. 
currently we are already getting all upcoming events when no filter is applied. 
currently when we click on all dates we get all upcoming events. When we click on past, we get all past events.
Desired Result:- when we click on all dates we get all events whether it is upcoming or past. When we click on All upcoming, we get all upcoming event(events that we get now when we reload explore page and on all dates filter). When we click past filter, we get all past filter. And most important, make all upcoming as pre-selected.

**Comment by iamareebjamal** [comment]
@Soumi150 Please complete the PR or close it

**Comment by Soumi150** [comment]
@iamareebjamal @sachinchauhan2889 Can we implement past and upcoming filters with a time frame like past 5 years or next 5 years .That can be done using moment js .We can then include past and upcoming within switch statement and all dates filter will then display both past and upcoming events.

**Pull request closed without merging**